src/audio/component.c

```diff
@@ -159,14 +159,23 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 		}
 		break;
 	case COMP_TRIGGER_STOP:
+		if (dev->state == COMP_STATE_ACTIVE) {
```

Comment:
It will fail for stop after pause, which is a valid transition.

Reply:
good point.